vignettes/getting-started-with-teal.Rmd

```diff
@@ -36,52 +36,36 @@ library(teal)
 app <- init(
   data = teal_data(IRIS = iris, MTCARS = mtcars),
   modules = example_module(),
-  header = "My first teal application"
+  title = "My first teal application",
+  header = h3("My first teal application"),
+  footer = div(a("Powered by teal", href = "https://insightsengineering.github.io/teal/latest-tag/"))
 )
 
 if (interactive()) {
   shinyApp(app$ui, app$server)
 }
 ```
+
 <img src="images/example_app.png" alt="Example application" style="width: 100%;"/>
 
-As shown in the image above, this application consists of several distinct areas:
+Every `teal` application is composed of following elements, which all can be controlled by app developer by passing arguments to `init` function:
 
-* <span style="color: #FB6251;">Application Header</span> (title at the top): is the title of the application.
-* <span style="color: #3A88FE;">Teal Modules</span> (bar under the title): in this case a simple module named "example teal module".
-* <span style="color: #BF37F3;">Encoding Panel</span> (panel on the left hand side): Module specific UI components, in this case a drop-down to select a dataset name.
-* <span style="color: #96D35F;">Main Output Panel</span> (panel on the middle): The outputs of the module, for this example module the chosen dataset is displayed.
+* <span style="color: #9f84ff;">Application Title</span> (browser's tab title): is the title of the application.
+* <span style="color: #FB6251;">Application Header and Footer</span> (The top and the bottom of the app): any content to be put on top and on the bottom of the application.
+* <span style="color: #3A88FE;">Teal Modules</span> (tabs under the header): tab for each module included in the application. In this case a simple module named "example teal module".
+* <span style="color: #96D35F;">Module Content</span> (panel on the middle): The outputs of the module, for this example side bar panel with encodings and module the chosen dataset is displayed.
```

Comment:
```suggestion
* <span style="color: #9f84ff;">Application Title</span> (browser's tab title): This is the title of the application.
* <span style="color: #FB6251;">Application Header and Footer</span> (the top and the bottom of the app): Any content to be placed at the top and bottom of the application.
* <span style="color: #3A88FE;">Teal Modules</span> (tabs under the header): A tab for each module included in the application. In this case, a simple module named "example teal module."
* <span style="color: #96D35F;">Module Content</span> (panel in the middle): The outputs of the module. For this example, a sidebar panel with encodings and the chosen dataset is displayed.
```

Reply:
I was thinking about changing this screenshot into simple wireframe. If we keep making screenshots we will have to update them anytime styling changes

Reply:
Please see again and let me know what you think. Editable file will stay in the package so each time it will be possible to change and rerender `svg`

 
![teal-app-components](https://github.com/insightsengineering/teal/assets/6959016/589a60ef-9398-436c-98b2-dd6d740918bb)

Reply:
I love the wireframe idea. Let's do this! Would be easier to maintain as well.

Reply:
I'll see your diagram and raise another diagram ;)
One thing that I liked about the previous screenshot was that it showed the output of the app code that we have and it's quite natural for someone to look for the output just below the code.
And, I agree that the new diagram is simpler along with the simplification of the components.

Why not have the best of both worlds by showing the diagram of the app output when this image is hovered?
I added a commit to implement this, please let me know what you think.
![Screenshot 2024-01-23 at 9 40 56 AM](https://github.com/insightsengineering/teal/assets/49812166/cf663124-7638-4ce6-993b-59e563794343)
![Screenshot 2024-01-23 at 9 40 50 AM](https://github.com/insightsengineering/teal/assets/49812166/64df8037-7d84-4d1e-92e5-ee48c40dea30)

Reply:
I like the idea of showing example when hovered but I think it defeat the purpose of changing the snapshot approach to a wireframe. 

For example, if we change the styling of the app, doesn't that mean we have to renew the screenshot for the example? This means, we now have to maintain the wireframe and the screenshot example. Felt like we're adding more work to ourselves.

Maybe we can refer user to teal.gallery to see example?

Reply:
It's a good idea to add `teal.gallery` and TLG/Biomarker Catalogs at the end. I'll add them shortly.

It still solves the aspect ratio issue while taking a screenshot. Moreover, this diagram is just "representative" and is just here to simulate how it looks. I still believe this is more maintainable than a screenshot (just a little more work though). In the future, when a theme would be changed we just need to change the diagram and export it, while in the case of a screenshot "someone" would have to take the same aspect ratio screenshot to make the hover be seamless and it would be challenging.

Reply:
Thanks @vedhav 
Let's just stick with one workflow, the diagram.
I think what we have now is good.